cli/azd/pkg/infra/provisioning/provider.go

```diff
@@ -135,13 +136,24 @@ func NewProvider(
 	projectPath string,
 	infraOptions Options,
 	prompters Prompters,
+	alphaFeatureManager *alphafeatures.AlphaFeatureManager,
 ) (Provider, error) {
 	var provider Provider
 
 	if infraOptions.Provider == "" {
 		infraOptions.Provider = Bicep
 	}
 
+	if alphafeatures.IsAlphaKey(string(infraOptions.Provider)) {
```

Comment:
I can see what you're going for, but I think we can invert and simplify this. I'm taking inspiration from feature-expose libraries I've worked a lot with in the past.

Application owners should just have to do this:

`features.IsEnabled(<my feature key>)`

and so, we would have:

in `features.go`:

```golang
type Key string
const Terraform = Key("terraform")
```

here:

```golang
if infraOptions.Provider == Terraform {
    if !features.IsEnabled(features.Terraform) {
        // return err
    }
}
```

Reply:
> if infraOptions.Provider == Terraform {

I don't like the fact that the provisioning provider is aware about terraform is an alpha feature and it has to check if it is enabled. That's a very particular fact.  I prefer going for the general case, where any provider-name can be an alpha feature and the `provisioning provider` should just check any name.

At least in this way, we have the check and validation for any new provider that is added as alpha feature.

Reply:
> I don't like the fact that the provisioning provider is aware about terraform is an alpha feature and it has to check if it is enabled. That's a very particular fact. I prefer going for the general case, where any provider-name can be an alpha feature and the provisioning provider should just check any name.

I think if you think a little deeper, that may not be the correct behavior.

An application feature should know about the feature flag it uses to communicate to the user. When we create the `terraform` application code, we would have decided on what the feature flag should be. It should not be coupled directly to the provider's name. In fact, we should allow that the feature flag is called `infra.provider` or something different. 

You can also consider what would have happened if I was creating the `terraform` code for the first time. It would feel natural for me to define in `features.go` the name of the feature, and I would have added an `if feature.IsEnabled(myFeature)`.

Reply:
> It should not be coupled directly to the provider's name.

It is a tradeoff between styles...  I thought about creating a mapping between the `FeatureId` (from the features.yaml) and the `Application Constant` that is used to check if it is enabled...  But, in most cases, it will be the same.  And also, if feels like those type of `conventions` are classic to go. For example, the `_test` suffix for file names, the lower/upper case for exporting, etc.

In this case, for adding a provisioning provider as `alpha`, we are introducing a convention where all you need to do is make the name of the provider match the featureId in the alpha list.  Then `azd` will automatically handle the rest. I will add a document about `how to add a new provisioning provider` as a bonus here, similar to the doc about adding a new command.

When it comes the time to add a new command as alpha, we can define if we want to create a similar convention to make it easier to mark commands as alpha or non alpha without leaking the details about it around the code implementation.